tests/bsim/bluetooth/host/gatt/sc_indicate/src/bs_bt_utils.c

```diff
@@ -0,0 +1,189 @@
+/**
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+
+#include <zephyr/bluetooth/gap.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/hci_types.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(test_utils, LOG_LEVEL_DBG);
+
+#include "bs_bt_utils.h"
+
+struct bt_conn *g_conn;
+DEFINE_FLAG(flag_is_connected);
+
+void wait_connected(void)
+{
+	LOG_DBG("Wait for connection...");
+	WAIT_FOR_FLAG(flag_is_connected);
+}
+
+void wait_disconnected(void)
+{
+	LOG_DBG("Wait for disconnection...");
+	WAIT_FOR_FLAG_UNSET(flag_is_connected);
+}
+
+static void security_changed(struct bt_conn *conn, bt_security_t level, enum bt_security_err err)
+{
+	LOG_DBG("security changed");
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	UNSET_FLAG(flag_is_connected);
+}
+
+struct bt_conn *get_g_conn(void)
+{
+	return g_conn;
+}
+
+void clear_g_conn(void)
+{
+	struct bt_conn *conn;
+
+	conn = g_conn;
+	g_conn = NULL;
+	ASSERT(conn, "Test error: no g_conn!\n");
+	bt_conn_unref(conn);
+}
+
+static void connected(struct bt_conn *conn, uint8_t err)
+{
+	ASSERT((!g_conn || (conn == g_conn)), "Unexpected new connection.");
```

Comment:
Suggest to use `if` instead of ASSERT as ASSERT can be disabled (and I guess we always want this check?)

Reply:
Oh. I don't redefining a common macro like `ASSERT` is a good idea either :D 

```
if (expr) {
    FAIL(...);
}
```

Isn't a large chunk of code to have, and it makes it very clear what is going on. 

If you want to keep it as a macro like above, then I suggest to rename it `BSIM_ASSERT` or `BS_BT_UTIL_ASSERT` or something like that, so it isn't confused with the _actual_ `ASSERT`

Reply:
You're right, I will rename it.